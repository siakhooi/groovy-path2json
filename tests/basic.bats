setup() {
    load 'common-setup'
    common_setup

    outputExpected=$BATS_TEST_FILENAME.$BATS_TEST_DESCRIPTION.expected.json
}

@test "basic" {
    tempDir=$(mktemp -d)
    cd "$tempDir"
    run path2json

    sed "s:TEMPDIR:${tempDir}:" "$outputExpected" | assert_output
}
@test "subdirectories" {
    tempDir=$(mktemp -d)
    cd "$tempDir"
    mkdir -p a/b/c
    touch a/b/c/d
    touch a/b/e
    run path2json

    sed "s:TEMPDIR:${tempDir}:g" "$outputExpected" | assert_output
}
