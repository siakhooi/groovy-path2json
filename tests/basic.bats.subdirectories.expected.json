{"creationTime":"2024-12-10T11:32:02.945112321Z","creationTimeInMillis":1733830322945,"lastAccessTime":"2024-12-10T11:32:02.945112321Z","lastAccessTimeInMillis":1733830322945,"lastModifiedTimeInMillis":1733830322945,"isDirectory":true,"isRegularFile":false,"path":"TEMPDIR","isSymbolicLink":false,"lastModifiedTime":"2024-12-10T11:32:02.945112321Z","isOther":false,"children":[{"creationTime":"2024-12-10T11:32:02.945112321Z","creationTimeInMillis":1733830322945,"lastAccessTime":"2024-12-10T11:32:02.945112321Z","lastAccessTimeInMillis":1733830322945,"lastModifiedTimeInMillis":1733830322945,"isDirectory":true,"isRegularFile":false,"path":"TEMPDIR/a","isSymbolicLink":false,"lastModifiedTime":"2024-12-10T11:32:02.945112321Z","isOther":false,"children":[{"creationTime":"2024-12-10T11:32:02.945112321Z","creationTimeInMillis":1733830322945,"lastAccessTime":"2024-12-10T11:32:02.945112321Z","lastAccessTimeInMillis":1733830322945,"lastModifiedTimeInMillis":1733830322955,"isDirectory":true,"isRegularFile":false,"path":"TEMPDIR/a/b","isSymbolicLink":false,"lastModifiedTime":"2024-12-10T11:32:02.955152141Z","isOther":false,"children":[{"creationTime":"2024-12-10T11:32:02.955152141Z","creationTimeInMillis":1733830322955,"lastAccessTime":"2024-12-10T11:32:02.955152141Z","lastAccessTimeInMillis":1733830322955,"lastModifiedTimeInMillis":1733830322955,"isDirectory":false,"isRegularFile":true,"path":"TEMPDIR/a/b/e","isSymbolicLink":false,"lastModifiedTime":"2024-12-10T11:32:02.955152141Z","isOther":false,"children":[]},{"creationTime":"2024-12-10T11:32:02.945112321Z","creationTimeInMillis":1733830322945,"lastAccessTime":"2024-12-10T11:32:02.945112321Z","lastAccessTimeInMillis":1733830322945,"lastModifiedTimeInMillis":1733830322955,"isDirectory":true,"isRegularFile":false,"path":"TEMPDIR/a/b/c","isSymbolicLink":false,"lastModifiedTime":"2024-12-10T11:32:02.955152141Z","isOther":false,"children":[{"creationTime":"2024-12-10T11:32:02.955152141Z","creationTimeInMillis":1733830322955,"lastAccessTime":"2024-12-10T11:32:02.955152141Z","lastAccessTimeInMillis":1733830322955,"lastModifiedTimeInMillis":1733830322955,"isDirectory":false,"isRegularFile":true,"path":"TEMPDIR/a/b/c/d","isSymbolicLink":false,"lastModifiedTime":"2024-12-10T11:32:02.955152141Z","isOther":false,"children":[]}]}]}]}]}